### PR TITLE
Scaffolding setup now supports Firebird 2.5

### PR DESCRIPTION
**Pull request by SJFriedl**
This adds scaffolding support for versions of Firebird < 3.0 (tested on 2.5) by eliding the `rdb$IDENTITY_TYPE` column if the version is too old - it uses the same logic as found in the `FbColumns.cs` support elsewhere.

This works well enough to create the raft of files representing each table, but it hasn't be tested for anything beyond that as Jiri has said other updates need to be done. So this is only an incremental step.

### CODE REVIEW

File: Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs [hunk 1]
```diff
@@ -192,11 +193,20 @@ ORDER BY
 
 		private void GetColumns(DbConnection connection, IReadOnlyList<DatabaseTable> tables, Func<string, string, bool> tableFilter)
 		{
+			var identity_type = "null";
+
+			if (connection.State == ConnectionState.Open)
```
Comment (cincuranet) [inline Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs:198]: At this point connection is known to be open.

File: Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs [hunk 2]
```diff
@@ -192,11 +193,20 @@ ORDER BY
...
+			{
+				var serverVersion = FbServerProperties.ParseServerVersion(connection.ServerVersion);
```
Comment (cincuranet) [inline Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs:200]: I would put this into `DatabaseModel Create(DbConnection connection, DatabaseModelFactoryOptions options)` method and create a property for it.

File: Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs [hunk 3]
```diff
@@ -192,11 +193,20 @@ ORDER BY
...
+				if (serverVersion.Major >= 3)
```
Comment (cincuranet) [inline Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs:201]: Store `MajorVersionNumber` directly, to make it easier for future.

**Review by cincuranet — CHANGES_REQUESTED** [review]

**Comment by cincuranet** [comment]
@SJFriedl Are you going to finish this PR?

**Comment by cincuranet** [comment]
Done via faedc6e7.

**Comment by lvalgimigli** [comment]
I have this error when scaffolding a 2.5 Firebird database:

**Comment by lvalgimigli** [comment]
I have this error when scaffolding a 2.5 Firebird database: Could not find type mapping for column 'XXXXX' with data type 'NUMERIC(4,2)'. Skipping column.

**Comment by mrotteveel** [comment]
Please report bugs by creating new issues, or if you're asking for help, ask on [firebird-net-provider](https://groups.google.com/g/firebird-net-provider)

**Comment by cincuranet** [comment]
@mrotteveel He did: https://github.com/FirebirdSQL/NETProvider/issues/1142.

**Pull request closed without merging**